### PR TITLE
[FEATURE] Introduced patch var in manager

### PR DESCRIPTION
**Pull request by turbo-ele**

### CODE REVIEW

File: Order/OrderManager.php [hunk 1]
```diff
@@ -693,6 +672,161 @@ function (&$order) use ($locale) {
     }
 
     /**
+     * @param array $itemData
+     * @param string $locale
+     * @param int $userId
+     * @param Order $order
+     *
+     * @return mixed
```
Comment (christianbader) [inline Order/OrderManager.php:680]: sure?

Comment (turbo-ele) [inline Order/OrderManager.php:680]: yes

File: Order/OrderManager.php [hunk 2]
```diff
@@ -693,6 +672,161 @@ function (&$order) use ($locale) {
...
+
+    /**
+     * @param ItemInterface $item
+     * @param array $itemData
+     * @param string $locale
+     * @param int $userId
+     *
+     * @return null|\Sulu\Bundle\Sales\CoreBundle\Api\Item
+     */
+    public function updateItem($item, $itemData, $locale, $userId)
```
Comment (christianbader) [inline Order/OrderManager.php]: ItemInterface $item

File: Order/OrderManager.php [hunk 3]
```diff
@@ -693,6 +672,161 @@ function (&$order) use ($locale) {
...
+    {
+        return $this->itemManager->save($itemData, $locale, $userId, $item);
+    }
+
+    /**
+     * @param ItemInterface $item
+     * @param OrderInterface $order
```
Comment (christianbader) [inline Order/OrderManager.php:704]: missing parameter

File: Order/OrderManager.php [hunk 4]
```diff
@@ -693,6 +672,161 @@ function (&$order) use ($locale) {
...
+            // delete item
+            $this->em->remove($item);
+        }
+    }
+
+    /**
+     * Get order item by id and checks if item belongs to the order
+     *
+     * @param int $itemId
+     * @param OrderInterface $order
```
Comment (christianbader) [inline Order/OrderManager.php:721]: missing parameter

File: Order/OrderManager.php [hunk 5]
```diff
@@ -942,20 +1068,6 @@ private function addContactRelation(array $data, $dataKey, $addCallback)
...
-     * @return mixed
-     */
-    private function getProperty(array $data, $key, $default = null)
-    {
-        return array_key_exists($key, $data) ? $data[$key] : $default;
-    }
-
-    /**
      * @param $data
      * @param Order $order
```
Comment (christianbader) [inline Order/OrderManager.php:1077]: @param missing

File: Order/OrderManager.php [hunk 6]
```diff
@@ -1005,7 +1117,7 @@ private function setTermsOfDelivery($data, Order $order)
      *
      * @return null|object
      */
-    private function setTermsOfPayment($data, Order $order)
+    private function setTermsOfPayment($data, Order $order, $patch = false)
```
Comment (christianbader) [inline Order/OrderManager.php:1127]: add patch to @param

File: Order/OrderManager.php [hunk 7]
```diff
@@ -1040,13 +1152,13 @@ private function setTermsOfPayment($data, Order $order)
     /**
      * @param array $data
      * @param Order $order
-     *
-     * @throws Exception\MissingOrderAttributeException
-     * @throws Exception\OrderDependencyNotFoundException
+     * @param bool $patch
      *
      * @return null|object
```
Comment (christianbader) [inline Order/OrderManager.php:1166]: move down

**Comment by christianbader** [comment]
else ack

**Pull request merged**